facebookresearch/ParlAI
# Adding features into the human+model turn annotation task

**Patch description**
Bring the public version of this MTurk task up to feature parity with the currently used internal version.

Added features:
- With `--blocklist-path` flag, ingests lists of users to soft-block
- With `--check-acceptability` flag, checks acceptability of Turker's messages
- With `--include-persona` flag, shows personas to the bot
- With `--conversation-start-mode` flag, starts a conversation with two previous utterances
- Saves all complex UI text and params as HTML or JSON files for easier changing of text via the command line
- Two ways to run task: command-line or launch script
- Clean + detailed README detailing both ways to run task

**Testing steps**
- Unit tests for both the chat and onboarding worlds: `pytest tests/nightly/gpu/test_crowdsourcing.py`
- Manual check that all UI elements defined by flags are being loaded properly
- Tested examples of running sandbox HITs with the launch script + CLI command
